### PR TITLE
BUG: Fix config loads to avoid KeyError

### PR DESCRIPTION
**Pull request by pllim**
Keep seeing KeyError on astropy-bot log still.

(PR from GitHub UI, so no fork, sorry!)

### CODE REVIEW

**Comment by pllim** [comment]
Fixing this failed 2 tests. Any advise how to fix the tests? 🤷

**Comment by astrofrog** [comment]
Should we return an empty config in the else case?

**Comment by Cadair** [comment]
I am not sure that's a good idea either, as there is at least some logic somewhere for falling back to an old toolname in the config.

**Comment by Cadair** [comment]
I think the issue is this block: https://github.com/OpenAstronomy/baldrick/blob/master/baldrick/github/github_api.py#L129-L140 expects an error.

**Comment by Cadair** [comment]
I guess that should just be patched to accept None rather than be a try block?

**Comment by pllim** [comment]
Okay. What about now?

File: baldrick/github/github_api.py [hunk 1]
```diff
@@ -126,17 +126,17 @@ def get_repo_config(self, branch='master', path_to_file='pyproject.toml'):
...
 
         if file_content:
-            try:
-                repo_config = loads(file_content, tool=current_app.bot_username)
-                logger.trace(f"Got the following config from {self.repo}@{branch}: {repo_config}")
-            except Exception:
+            repo_config = loads(file_content, tool=current_app.bot_username)
+            logger.trace(f"Got the following config from {self.repo}@{branch}: {repo_config}")
+            if not repo_config:
+                repo_config = {}
```
Comment (Cadair) [inline baldrick/github/github_api.py:132]:
```suggestion
            repo_config = loads(file_content, tool=current_app.bot_username) or {}
            logger.trace(f"Got the following config from {self.repo}@{branch}: {repo_config}")
```

File: baldrick/github/github_api.py [hunk 2]
```diff
@@ -126,17 +126,17 @@ def get_repo_config(self, branch='master', path_to_file='pyproject.toml'):
...
                 logger.exception(
                     f"Failed to load config in {self.repo} on branch {branch}, despite finding a pyproject.toml file.")
 
             if getattr(current_app, "fall_back_config", None):
-                try:
-                    fallback_config = loads(file_content, tool=current_app.fall_back_config)
-                except Exception:
+                fallback_config = loads(file_content, tool=current_app.fall_back_config)
+                if not fallback_config:
+                    fallback_config = {}
```
Comment (Cadair) [inline baldrick/github/github_api.py:139]:
```suggestion
                fallback_config = loads(file_content, tool=current_app.fall_back_config) or {}
```

**Review by Cadair — APPROVED** [review]

**Comment by pllim** [comment]
Thanks! Now, how do I make `astropy-bot` pick this up?

**Pull request merged**